src/controllers/dynakube/version/healthcheck.go

```diff
@@ -0,0 +1,79 @@
+package version
+
+import (
+	"context"
+	"time"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/src/registry"
+	containerv1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DefaultHealthConfigInterval    = time.Second * 10
+	DefaultHealthConfigStartPeriod = time.Second * 1200
+	DefaultHealthConfigTimeout     = time.Second * 30
+	DefaultHealthConfigRetries     = 3
+)
+
+// Constructor setting default values for docker image HealthConfig
+func newHealthConfig() *containerv1.HealthConfig {
+	healthConfig := &containerv1.HealthConfig{}
+	healthConfig.Test = []string{}
+	healthConfig.Interval = DefaultHealthConfigInterval
+	healthConfig.StartPeriod = DefaultHealthConfigStartPeriod
+	healthConfig.Timeout = DefaultHealthConfigTimeout
+	healthConfig.Retries = DefaultHealthConfigRetries
+
+	return healthConfig
+}
+
+func SetOneAgentHealthcheck(ctx context.Context, apiReader client.Reader, registryClient registry.ImageGetter, dynakube *dynatracev1beta1.DynaKube, imageUri string) error {
+	imageInfo, err := PullImageInfo(ctx, apiReader, registryClient, dynakube, imageUri)
+	if err != nil {
+		return errors.WithMessage(err, "error pulling image info")
+	}
+
+	configFile, err := (*imageInfo).ConfigFile()
+	if err != nil {
+		return errors.WithMessage(err, "error reading image config file")
+	}
+
+	// Healthcheck.Test values from go-containerregistry documentation:
+	// {} : inherit healthcheck
+	// {"NONE"} : disable healthcheck
+	// {"CMD", args...} : exec arguments directly
+	// {"CMD-SHELL", command} : run command with system's default shell
+	if configFile.Config.Healthcheck != nil && len(configFile.Config.Healthcheck.Test) > 0 {
+		var testCommand []string
+
+		switch configFile.Config.Healthcheck.Test[0] {
+		case "CMD-SHELL":
+			testCommand = []string{"/bin/sh", "-c"}
+			testCommand = append(testCommand, configFile.Config.Healthcheck.Test[1:]...)
+		case "CMD":
+			testCommand = configFile.Config.Healthcheck.Test[1:]
+		}
+
+		if len(testCommand) > 0 {
+			healthConfig := newHealthConfig()
+			healthConfig.Test = testCommand
+			if configFile.Config.Healthcheck.Interval != 0 {
```

Comment:
I think since we set defaults early on line 61, we should compare with defailts:
E.g.
```
 configFile.Config.Healthcheck.Interval != DefaultHealthConfigInterval
```

Reply:
nvm I misread it;